pkg/kubelet/qos/policy.go

```diff
@@ -62,13 +65,44 @@ func GetContainerOOMScoreAdjust(pod *v1.Pod, container *v1.Container, memoryCapa
 	// which use more than their request will have an OOM score of 1000 and will be prime
 	// targets for OOM kills.
 	// Note that this is a heuristic, it won't work if a container has many small processes.
-	memoryRequest := container.Resources.Requests.Memory().Value()
-	if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+	containerMemoryRequest := container.Resources.Requests.Memory().Value()
+
+	isInPlacePodVerticalScaling := utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling)
+	if isInPlacePodVerticalScaling {
 		if cs, ok := podutil.GetContainerStatus(pod.Status.ContainerStatuses, container.Name); ok {
-			memoryRequest = cs.AllocatedResources.Memory().Value()
+			containerMemoryRequest = cs.AllocatedResources.Memory().Value()
+		}
+	}
+	oomScoreAdjust := 1000 - (1000*containerMemoryRequest)/memoryCapacity
+
+	// adapt the sidecarContainer memoryRequest for OOM ADJ calculation
+	// calculate the oom score adjustment based on: min-memory( currentSideCarContainer , min-memory(regular containers) ) .
+	if utilfeature.DefaultFeatureGate.Enabled(features.SidecarContainers) && isSidecarContainer(pod, container) {
+		// check min resources with regular containers
+		minResourceList := resourcehelper.MinRegularContainerResourceList(*pod, resourcehelper.PodResourcesOptions{
```

Comment:
this is over generalized. Hard to read. You only need the minimal memory here.

Reply:
done